File: lib/connection_status.dart
enum ConnectionStatus {
  connecting,
  connected,
  disconnecting,
  disconnected,
  unknown;

  factory ConnectionStatus.fromString(String s) {
    return ConnectionStatus.values
        .firstWhere((v) => v.name == s, orElse: () => ConnectionStatus.unknown);
  }
}

File: darwin/Classes/ConnectionStatus.swift
import Foundation
import NetworkExtension

/// ConnectionStatus returned to Dart
enum ConnectionStatus: String {
    case connected
    case disconnected
    case connecting
    case disconnecting
    case unknown

    static func fromNEVPNStatus(status: NEVPNStatus) -> ConnectionStatus {
        switch status {
        case .connected: return ConnectionStatus.connected
        case .disconnected: return ConnectionStatus.disconnected
        case .connecting: return ConnectionStatus.connecting
        case .disconnecting: return ConnectionStatus.disconnecting
        default: return ConnectionStatus.unknown
        }
    }

    func string() -> String {
        rawValue
    }
}
